modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/SortedIndexStorage.java

```diff
@@ -56,13 +57,20 @@ public interface SortedIndexStorage extends IndexStorage {
      *      {@code null} means unbounded.
      * @param flags Control flags. {@link #GREATER} | {@link #LESS} by default. Other available values
      *      are {@link #GREATER_OR_EQUAL}, {@link #LESS_OR_EQUAL}.
+     * @param onlyBuiltIndex {@code True} if need to scan only from the built index.
      * @return Cursor with fetched index rows.
      * @throws IllegalArgumentException If backwards flag is passed and backwards iteration is not supported by the storage.
+     * @throws StorageException If failed to read data.
+     * @throws IndexNotBuiltException If the index has not yet been built and {@code onlyBuiltIndex} = {@code true}.
+     * @throws InconsistentIndexStateException If the index is in a readable status, but the index is not built.
      */
+    // TODO: IGNITE-22039 Implement for onlyBuiltIndex == false throw an InconsistentIndexStateException if the index is not in a readable
+    //  status and write tests
     PeekCursor<IndexRow> scan(
             @Nullable BinaryTuplePrefix lowerBound,
             @Nullable BinaryTuplePrefix upperBound,
-            @MagicConstant(flagsFromClass = SortedIndexStorage.class) int flags
+            @MagicConstant(flagsFromClass = SortedIndexStorage.class) int flags,
+            boolean onlyBuiltIndex
```

Comment:
The usage of `true` and `false` is very unequal in the project. `false` is only used once, and it's in a very special case, while `true` is used 'normally'. How about making 2 methods: one would be a normal `scan()` with the same parameters that it had before this change (and its implementation will imply `onlyBuiltIndex == true`); and another like `tolerantScan()` that would tolerate inconsistencies (again, without the boolean parameter, instead it would just imply `onlyBuiltIndex == false`).

This would allow us to avoid polluting the code with that `true` everywhere we want to just scan the index without bringing any confusion, and the special case would be handled by the special method.

Reply:
fix it